src/EnergyPlus/InternalHeatGains.cc

```diff
@@ -472,53 +303,41 @@ namespace InternalHeatGains {
                                                                          state.dataIPShortCut->cAlphaFieldNames,
                                                                          state.dataIPShortCut->cNumericFieldNames);
 
-                for (Item1 = 1; Item1 <= state.dataHeatBal->PeopleObjects(Item).NumOfZones; ++Item1) {
-                    ++Loop;
-                    if (!state.dataHeatBal->PeopleObjects(Item).ZoneListActive) {
-                        state.dataHeatBal->People(Loop).Name = AlphaName(1);
-                        state.dataHeatBal->People(Loop).ZonePtr = state.dataHeatBal->PeopleObjects(Item).ZoneOrZoneListPtr;
-                    } else {
-                        CheckCreatedZoneItemName(
-                            state,
-                            RoutineName,
-                            CurrentModuleObject,
-                            state.dataHeatBal->Zone(state.dataHeatBal->ZoneList(state.dataHeatBal->PeopleObjects(Item).ZoneOrZoneListPtr).Zone(Item1))
-                                .Name,
-                            state.dataHeatBal->ZoneList(state.dataHeatBal->PeopleObjects(Item).ZoneOrZoneListPtr).MaxZoneNameLength,
-                            state.dataHeatBal->PeopleObjects(Item).Name,
-                            state.dataHeatBal->People,
-                            Loop - 1,
-                            state.dataHeatBal->People(Loop).Name,
-                            errFlag);
-                        state.dataHeatBal->People(Loop).ZonePtr =
-                            state.dataHeatBal->ZoneList(state.dataHeatBal->PeopleObjects(Item).ZoneOrZoneListPtr).Zone(Item1);
-                        if (errFlag) ErrorsFound = true;
-                    }
-
-                    state.dataHeatBal->People(Loop).NumberOfPeoplePtr = GetScheduleIndex(state, AlphaName(3));
+                // Create one People instance for every space associated with this People input object
+                auto &thisPeopleInput = state.dataHeatBal->PeopleObjects(peopleInputNum);
+                for (int Item1 = 1; Item1 <= thisPeopleInput.numOfSpaces; ++Item1) {
```

Comment:
Loop over all spaces for a given input object (as determined by `setupIHGZonesAndSpaces`.